Sofa/Component/Constraint/Projective/src/sofa/component/constraint/projective/LinearMovementConstraint.inl

```diff
@@ -385,6 +385,21 @@ void LinearMovementConstraint<DataTypes>::applyConstraint(const core::Mechanical
     }
 }
 
+template <class TDataTypes>
+void LinearMovementConstraint<TDataTypes>::applyConstraint(
+    sofa::core::behavior::ZeroDirichletCondition* matrix)
+{
+    constexpr unsigned int N = Deriv::size();
+
+    for (const auto index : m_indices.getValue())
+    {
+        for (unsigned int c = 0; c < N; ++c)
+        {
+            matrix->discardRowCol(N * index + c, N * index + c);
```

Comment:
does it make any optimisation to compute N*index before the loop on c ?

Reply:
Normally, this kind of optimisation can be made by the compiler. But I did not check